tests/ui/traits/new-solver/gat-bound-fail.rs

```diff
@@ -0,0 +1,18 @@
+// compile-flags: -Ztrait-solver=next
+
+trait Foo {
+    type Gat<T: Bar>;
+}
+
+impl Foo for () {
+    type Gat<T: Bar> = ();
+}
+
+trait Bar {}
+
+impl Bar for i32 {}
+
+fn main() {
+    let _: <() as Foo>::Gat<()> = ();
```

Comment:
Actually, upon second thought, this test may not actually be checking that GAT bounds are being checked during confirmation, but just checking path instantiation code in HIR typeck... 

Otherwise, I have no idea how this error message is turning into `(): Bar` not satisfied.